colossalai/quantization/fp8.py

```diff
@@ -101,10 +115,15 @@ def all_reduce_fp8(tensor: torch.Tensor, fp8_format="e4m3", op=ReduceOp.SUM, gro
     inp = ret.view(torch.uint8)
     input_chunks = list(torch.chunk(inp, world_size, dim=0))
     output_chunks = list(torch.chunk(torch.empty_like(inp), world_size, dim=0))
-    dist.all_to_all(output_chunks, input_chunks, group=group)
+    chunck_handle = dist.all_to_all(output_chunks, input_chunks, group=group, async_op=async_op)
     scale_list = [torch.ones(1, dtype=scale.dtype, device=input_device) for _ in range(world_size)]
-    dist.all_gather(scale_list, scale, group=group)
+    scale_handle = dist.all_gather(scale_list, scale, group=group, async_op=async_op)
     summed_out = torch.zeros_like(output_chunks[0]).to(input_type)
+
+    if async_op:
+        chunck_handle.wait()
+        scale_handle.wait()
```

Comment:
Why wait when async op is enabled?

Reply:
There are operations depends on the data from these two handles. Maybe there is a better way to do this?